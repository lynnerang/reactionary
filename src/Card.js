import React, { Component } from 'react';
import './Card.scss';

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showExample: false
    }
  }

  showExample = () => {
    this.setState({showExample: true});
  }

  closeExample = () => {
    this.setState({showExample: false});
  }

  render() {
    // const exampleId = this.props.cardData.id;
    const url = require(`./images/example28.png`);

    let popup = this.state.showExample && <div className='popup'>
        <img className='example-img' src={url} alt='visual example' onBlur={this.closeExample}/>
        <i className='fas fa-times popup-close-btn' onClick={this.closeExample}> close</i>
      </div>;
 
    let term = this.props.hasTerm ? this.props.cardData.term
    : 'What am I?';
    
    const deleteBtn = this.props.canDelete && <i className='fa fa-trash' onClick={this.props.deleteMyCard}></i>

    return (
      <>
      {popup}
        <article className='card'>
          {deleteBtn}        
          <div className='card-top'>
            <h4 className='term-header'>{term}</h4>
          </div>
          <div className='card-bottom'>
            <h4 className='description-header'>Description: </h4>
            <p className='description'>{this.props.cardData.definition}</p>
            <p className='example-link' onClick={this.showExample}>(View Example)</p>
          </div>
          </article>
        </>
    )
  }
}

export default Card;