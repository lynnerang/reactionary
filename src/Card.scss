@import './variables.scss';

.card {
  position: relative;
  height: 380px;
  min-width: 310px;
  max-width: 335px;
  background: rgb(236, 245, 244);
  border-radius: 10px;
  padding: 20px 30px;
  margin: 20px 20px 20px 20px;
  display: flex;
  flex-direction: column;
  box-shadow: 2px 2px 5px 2px rgba(211, 211, 211, 0.564);

  &:first-child {
    margin-left: 0px;
  }
}

.card-bottom {
  flex-grow: 1;
  padding-left: 5px;
}

.card-top {
  position: relative;
  text-align: center;
  background: rgba(255, 255, 255, 0.906);
  padding-top: 14px;
  padding-bottom: 10px;
  margin-bottom: 15px;
  border-radius: 3px;

  i {
    position: absolute;
    font-size: 18px;
    right: 30px;
    color: white;
  }
}

.term-header {
  margin: 0;
  color: $dark-gray;
  font-weight: 500;
  font-size: 21px;
}

.description {
  font-family: 'Open Sans', sans-serif;
  margin-bottom: 12px;
  font-size: 17px;
  line-height: 1.3;
}

h4 {
  font-weight: 400;
  margin-top: 10px;
  margin-bottom: 15px;
}

.example-link {
  font-size: 15px;
  font-weight: 400;
  font-family: 'Open Sans', sans-serif;
  color: $secondary-color;
  text-decoration: underline;

  &:hover {
    color: $blue;
  }
}

.description-header {
  font-size: 22px;
  color: $primary-color;
}

.popup {
  position: fixed;
  top: 0px;
  left: 0px;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(0, 0, 0, 0.7);
  z-index: 10;

  img {
    max-height: 80%;
    max-width: 80%;
  }

  div {
    padding: 30px;
    background: white;
    border-radius: 5px;
  }

  .popup-close-btn {
    position: absolute;
    top: 20px;
    right: 40px;
    z-index: 15;
    color: white;
    font-size: 25px;
    cursor: pointer;
  }
}

.fa-trash {
  position: absolute;
  bottom: 25px;
  right: 25px;
  color: rgba(128, 128, 128, 0.612);
  z-index: 3;

  &:hover {
    color: rgba(255, 0, 0, 0.413);
  }
}

.hidden {
  display: none;
}