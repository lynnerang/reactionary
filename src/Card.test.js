import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card.js';
import { shallow } from 'enzyme';

const mockData = {
  id: 2,
  term: "Compilers",
  fakeAnswers: ["Bundlers", "Package Managers"],
  definition: "These things take JavaScript code, transform it, and return it as JavaScript code in a different format.",
  example: "Babel is a popular example, and is most commonly used for React and transforms ES6 syntax into syntax that older browsers can interpret."
}

const props = {
  deleteMyCard: jest.fn(),
};

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <Card {...props}
      hasTerm={false} 
      canDelete={true} 
      cardData={mockData}
      />
      )
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card 
      hasTerm={false} 
      canDelete={true} 
      cardData={mockData}
      />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has one default state properties', () => {
    expect(wrapper.state()).toEqual({showExample: false});
  });

  it('should toggle the showExample state', () => {
    expect(wrapper.state('showExample')).toEqual(false);

    wrapper.instance().toggleExample();

    expect(wrapper.state('showExample')).toEqual(true);
  });

  it('can run a method to tell a parent component to delete it', () => {
    wrapper.instance().deleteCard();

    expect(props.deleteMyCard).toHaveBeenCalledTimes(1);

    expect(props.deleteMyCard).toHaveBeenCalledWith(mockData.id);
  });

});