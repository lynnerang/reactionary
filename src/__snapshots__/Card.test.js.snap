// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
<Fragment>
  <article
    className="card"
  >
    <i
      className="fa fa-trash"
      onClick={[Function]}
    />
    <div
      className="card-top"
    >
      <h4
        className="term-header"
      >
        What am I?
      </h4>
    </div>
    <div
      className="card-bottom"
    >
      <h4
        className="description-header"
      >
        Description: 
      </h4>
      <p
        className="description"
      >
        These things take JavaScript code, transform it, and return it as JavaScript code in a different format.
      </p>
      <p
        className="example-link"
        onClick={[Function]}
      >
        (View Example)
      </p>
    </div>
  </article>
</Fragment>
`;
